### PR TITLE
change default logger to dedicated one

### PR DESCRIPTION
**Pull request by cmpute**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Changed the default logger to `logging.getLogger("lightning")` as proposed here: https://github.com/PyTorchLightning/pytorch-lightning/issues/710#issuecomment-592973492
This modification won't make any difference to current users since the logging will propagate to root logger. However if the user want to distinguish the text log from pytorch-lightning from logs of other packages, they have an option now.

### CODE REVIEW

**Comment by Borda** [comment]
hey there, we have added GPU CI test, so could we kindly ask to rebase/merge master which will trigger these tests so we do not need to test it manually... Thx for your understanding :robot:

**Comment by cmpute** [comment]
Done!

File: pytorch_lightning/core/memory.py [hunk 1]
```diff
@@ -3,14 +3,14 @@
...
 import gc
-import logging as log
 import os
 import subprocess
 from subprocess import PIPE
 
 import numpy as np
 import torch
 
+from ..core import root_logger as log
```
Comment (awaelchli) [inline pytorch_lightning/core/memory.py]: the rest of the code base uses absolute imports. maybe change it to be consistent.

File: pytorch_lightning/trainer/training_io.py [hunk 2]
```diff
@@ -102,6 +102,7 @@
 import torch
 import torch.distributed as dist
 
+from pytorch_lightning.core import root_logger as log
```
Comment (awaelchli) [inline pytorch_lightning/trainer/training_io.py]: I think tests fail because this import is unused or something.

**Comment by cmpute** [comment]
Fixed

File: pytorch_lightning/callbacks/early_stopping.py [hunk 3]
```diff
@@ -5,13 +5,12 @@
...
 """
 
-import logging as log
 import warnings
 
 import numpy as np
 
 from .base import Callback
-
+from ..core import root_logger
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py]: use absolute imports
```suggestion
from pytorch_lightning.core import root_logger
```

File: pytorch_lightning/core/__init__.py [hunk 4]
```diff
@@ -312,6 +312,9 @@ def test_dataloader(self):
 
 """
 
+import logging
+root_logger = logging.getLogger("lightning")
```
Comment (Borda) [inline pytorch_lightning/core/__init__.py]: rather in `pytorch_lightning/__init__.py`

**Comment by cmpute** [comment]
Rebased & fixed

**Comment by Borda** [comment]
this bug shall be fixed in #1163

File: pytorch_lightning/__init__.py [hunk 5]
```diff
@@ -24,6 +24,9 @@
     # We are not importing the rest of the scikit during the build
     # process, as it may not be compiled yet
 else:
+    import logging
+    _logger = logging.getLogger("lightning")
```
Comment (Borda) [inline pytorch_lightning/__init__.py]:
```suggestion
    from logging import getLogger
    _logger = getLogger("lightning")
```
we had some problem with this importing logging in past

File: pytorch_lightning/profiler/profiler.py [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import cProfile
 import io
-import logging as log
 import pstats
 import time
 from abc import ABC, abstractmethod
 from collections import defaultdict
 from contextlib import contextmanager
+from pytorch_lightning import _logger as log
```
Comment (Borda) [inline pytorch_lightning/profiler/profiler.py]: the imports are usually structured into three groups, separated by blank line
1. build-in packages (e.g. time)
2. third party packages (e.g numpy)
3. this module - pytorch_lightning

Comment (cmpute) [reply pytorch_lightning/profiler/profiler.py]: Got it!

**Comment by cmpute** [comment]
About the docs, I didn't read into them carefully so I think I can make improvements after I'm more familiar with the framework

**Comment by Borda** [comment]
@cmpute could you pls update from master?

**Comment by Borda** [comment]
@cmpute could you pls update from master? we have fixed the failure in #1163

**Comment by cmpute** [comment]
Rebased

**Pull request merged**